imi/src/main/java/org/motechproject/nms/imi/service/impl/TargetFileServiceImpl.java

```diff
@@ -579,33 +872,66 @@ private void sendNotificationRequest(TargetFileNotification tfn) {
     @Transactional
     public void generateTargetFile(MotechEvent event) {
         LOGGER.debug(event.toString());
+        if(Boolean.valueOf(settingsFacade.getProperty(generateJhFile))) {
+            LOGGER.info("boolean:"+Boolean.valueOf(settingsFacade.getProperty(generateJhFile)));
+            TargetFileNotification[] tfn = generateObdFiles();
+            if (tfn != null) {
+                // Copy the OBD file from the local imi.local_obd_dir to the remote imi.local_obd_dir network share
+                ScpHelper scpHelper = new ScpHelper(settingsFacade);
+                for(TargetFileNotification t: tfn) {
+                    try {
+                        scpHelper.scpObdToRemote(t.getFileName());
+                    } catch (ExecException e) {
+                        String error = String.format("Error copying target file %s: %s", t.getFileName(),
+                                e.getMessage());
+                        LOGGER.error(error,e);
+                        fileAuditRecordDataService.create(new FileAuditRecord(
+                                FileType.TARGET_FILE,
+                                t.getFileName(),
+                                false,
+                                error,
+                                null,
```

Comment:
The code seems duplicate … please abstract it out and use single method with params